lib/actions/transform.js

```diff
@@ -12,36 +10,21 @@ const PASSTHROUGH_BODY_PROPERTY = 'elasticio';
  * @param cfg configuration that is account information and configuration field values
  */
 async function processAction(msg, cfg) {
-  const expression = cfg.expression;
-  const compiledExpression = jsonata(expression);
-  handlePassthrough(msg);
-  console.log('Evaluating expression="%s" on body=%j', expression, msg.body);
-  const result = compiledExpression.evaluate(msg.body);
-  console.log('Evaluation completed, result=%j', result);
+  const result = JsonataTransform.jsonataTransform(msg, cfg);
+
   if (result === undefined || result === null || Object.keys(result).length === 0) {
     return Promise.resolve();
   }
-  if (typeof result[Symbol.iterator] === 'function') {
+
+  if (typeof result[Symbol.iterator] === 'function' && !cfg.dontSplitArray) {
         // We have an iterator as result
     for (const item of result) {
       await this.emit('data', eioUtils.newMessageWithBody(item));
     }
     return Promise.resolve();
   }
-  return Promise.resolve(eioUtils.newMessageWithBody(result));
-}
 
-function handlePassthrough(message) {
-  if (message.passthrough && Object.keys(message.passthrough)) {
-    if (PASSTHROUGH_BODY_PROPERTY in message.body) {
-      throw new Error(`${PASSTHROUGH_BODY_PROPERTY} property is reserved \
-            if you are using passthrough functionality`);
-    }
-
-    message.body.elasticio = {};
-    Object.assign(message.body.elasticio, message.passthrough);
-  }
-  return message;
+  return Promise.resolve(eioUtils.newMessageWithBody(result));
```

Comment:
I see that you are copying the `Promise.resolve()` stuff that was already here but it probably makes sense as part of cleaning up this component to remove all the `Promise.resolve()` stuff and just return what is in the parentheses.